load.php

```diff
@@ -253,9 +263,56 @@ function perflab_load_active_and_valid_modules() {
 		require_once PERFLAB_PLUGIN_DIR_PATH . 'modules/' . $module . '/load.php';
 	}
 }
-
 perflab_load_active_and_valid_modules();
 
+/**
+ * Places the Performance Lab's object cache drop-in in the drop-ins folder.
+ *
+ * This only runs in WP Admin to not have any potential performance impact on
+ * the frontend.
+ *
+ * This function will short-circuit if the constant
+ * 'PERFLAB_DISABLE_OBJECT_CACHE_DROPIN' is set as true.
+ *
+ * @since n.e.x.t
+ */
+function perflab_maybe_set_object_cache_dropin() {
+	global $wp_filesystem;
+
+	// Bail if disabled via constant.
+	if ( defined( 'PERFLAB_DISABLE_OBJECT_CACHE_DROPIN' ) && PERFLAB_DISABLE_OBJECT_CACHE_DROPIN ) {
+		return;
+	}
+
+	// Bail if already placed.
+	if ( PERFLAB_OBJECT_CACHE_DROPIN_VERSION ) {
+		return;
+	}
+
+	// Bail if already attempted before timeout has been completed.
+	// This is present in case placing the file fails for some reason, to avoid
+	// excessively retrying to place it on every request.
+	$timeout = get_transient( 'perflab_set_object_cache_dropin' );
+	if ( false !== $timeout ) {
+		return;
+	}
+
+	if ( $wp_filesystem || WP_Filesystem() ) {
+		// If there is an actual object-cache.php file, rename it.
+		// The Performance Lab object-cache.php will still load it, so the
+		// behavior does not change.
+		if ( $wp_filesystem->exists( WP_CONTENT_DIR . '/object-cache.php' ) ) {
+			$wp_filesystem->move( WP_CONTENT_DIR . '/object-cache.php', WP_CONTENT_DIR . '/object-cache-orig.php' );
+		}
+
+		$wp_filesystem->copy( PERFLAB_PLUGIN_DIR_PATH . 'server-timing/object-cache.php', WP_CONTENT_DIR . '/object-cache.php' );
```

Comment:
@eugene-manuilov I think that's okay. The vast majority of sites are on a single server, so automatically setting that up benefits the majority. Also without this being automated, it would result in much lower usage just because expecting the end users to place that file may be a bit too much to ask as a baseline experience.

The feature is already built in a way that, if placing the file does not work (for whatever reason, real-world servers are configured in _weird_ ways sometimes), everything would keep working as expected, just without any early metrics reporting. So I'd argue that wouldn't be a major problem for sites on multiple servers either. Also, to be fair, probably the average site with multiple servers is a "professional" setup where that file couldn't be automatically placed anyway (writing to the filesystem to other than the `uploads` directory is often disabled).

Reply:
Ok, sounds good then. If this doesn't break anything on multi-server setups then it's fine with me.